tempodb/encoding/vparquet/wal_block.go

```diff
@@ -589,6 +594,10 @@ func (b *walBlock) Fetch(ctx context.Context, req traceql.FetchSpansRequest, opt
 		Results: &mergeSpansetIterator{
 			iters: iters,
 		},
+		Bytes: func() uint64 {
+			// size of walBlock, we assume flushed block size was read during query
+			return b.FlushedSize()
```

Comment:
metrics includes an entry called "total block bytes" which is different then "inspected bytes". i think we should report data retrieved from the backend for "inspected" and use this flushed size for "total block bytes".

Reply:
I need InspectedBytes to calculate query throughput down the line where TraceQL is executed. 

I am using an existing `Bytes` callback to pass one metric up the chain to caller of the Fetch, and added a TODO to do it in a better way.

since I can only pass one metric back using this callback, I am passing flushed size as an approximation for InspectedBytes.

This approximation is only for walBlock and backend Block is passing it's `TotalBytesRead` metric on `BackendReaderAt`, which is also kinda not ideal but works for now 🤞🏼 

I am going to drop few TODOs for me to come back and find a better solution.

Reply:
Why don't we just do a similar pattern as exists for blocks opened in object storage? They use the `BackendReaderAt` shim that records how many bytes are read. We can just make a `walReaderAt` that implements `io.ReaderAt` and counts the bytes read on the files?

I'm really struggling with returning such different values for the backend blocks and the wal. I'd also prefer not to add the `FlushedSize()` method to the `WALBlock` interface.

If we use a readerat shim like the backend does we can avoid both of these issues.

Reply:
added a reader in wal page, and using that to compute accurate info about bytes read.